content/platform-builder/custom-host/fromspec.md

```diff
@@ -0,0 +1,42 @@
+---
+title: "Building a Host from Spec"
+date: 2018-12-29T11:02:05+06:00
+weight: 5
+draft: false
+---
+
+In the preceding section of the documentation, we talked about how to create your own custom hosts that utilize
+the existing wasmcloud-host Rust crate. In this section, we'll talk about the specification and requirements for wasmCloud
+hosts written from scratch either for internal use or community contribution (e.g. a custom host for an `ESP32` microcontroller, etc).
+
+If you want to make your own wasmCloud host, you're likely doing so because you want to add a device or operating system that we don't support to the ecosystem, one that will be able to co-exist with first-party wasmCloud hosts and interact with the lattice. The following requirements for hosts will give you the recipe you need to create your own host.
+
+### Host Requirements (Internal)
+
+Internally, the host must perform the following activities:
+
+* Periodically send a [HealthRequest](https://github.com/wasmCloud/actor-interfaces/blob/main/actor-core/core.widl#L24) message to each running actor _and_ capability provider. These messages are to remain internal and are NOT to be transmitted across a lattice. It is up to the host as to what it does with the information it gleans from these health checks, but health checks _must_ be performed.
+* Properly dispatch the `BindActor` message to providers  
+* Properly dispatch the `RemoveActor` message to providers
+* Handle _local_ dispatch between actors and providers if the host supports internal providers
+* Handle _local_ dispatch between actors and actors
+* Expose some form of API that can be used to start and stop actors
+  * Actors must communicate with the host using the waPC specification. Custom hosts do not have to re-use existing waPC Rust crates, they only have to ensure that the host exposes the same waPC functions as the first-party wasmCloud host.
```

Comment:
I think a link here to the waPC spec would be useful, as this is a pretty significant part of the implementation

Reply:
done.